### PR TITLE
Fix intermittent test_profile_plot failure

### PR DESCRIPTION
**Pull request by mrocklin**
Fixes https://github.com/dask/distributed/issues/6454

cc @graingert for review if you have a moment.

### CODE REVIEW

**Comment by pentschev** [comment]
Wow, 100% pass now, haven't seen that in a while. Very nice! 😄

**Comment by mrocklin** [comment]
Merging in 24 hours if there are no comments

File: distributed/dashboard/tests/test_components.py [hunk 1]
```diff
@@ -25,9 +25,9 @@ def test_basic(Component):
 async def test_profile_plot(c, s, a, b):
     p = ProfilePlot()
     assert not p.source.data["left"]
-    await c.gather(c.map(slowinc, range(10), delay=0.05))
-    p.update(a.profile_recent)
-    assert len(p.source.data["left"]) >= 1
+    while not len(p.source.data["left"]):
+        await c.submit(slowinc, 1, pure=False, delay=0.1)
```
Comment (gjoseph92) [inline distributed/dashboard/tests/test_components.py:29]:
```suggestion
    await c.submit(slowinc, 1, pure=False, delay=0.1)
    while not len(p.source.data["left"]):
```
Is resubmitting necessary?

Comment (mrocklin) [reply distributed/dashboard/tests/test_components.py:29]: Yeah, if the profiler didn't run during the task then we're screwed I think.  We want to continuously run stuff until the profiler picks it up and that information arrives on the scheduler.

Comment (gjoseph92) [reply distributed/dashboard/tests/test_components.py:29]: Makes sense, thanks

**Review by gjoseph92 — APPROVED** [review]

**Comment by mrocklin** [comment]
Thanks for the review @gjoseph92

**Pull request merged**